ksml-integration-tests/src/test/java/io/axual/ksml/integration/ApicurioAvroSchemaRegistryIT.java

```diff
@@ -0,0 +1,332 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * =========================LICENSE_END==================================
+ */
+
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.clients.admin.AdminClientConfig;
+import org.apache.kafka.clients.admin.NewTopic;
+import org.apache.kafka.clients.consumer.ConsumerConfig;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.io.TempDir;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.output.Slf4jLogConsumer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+import org.testcontainers.kafka.KafkaContainer;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import java.io.File;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Properties;
+import java.util.concurrent.ExecutionException;
+
+import lombok.extern.slf4j.Slf4j;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * KSML Integration Test with Apicurio AVRO and Apicurio Schema Registry that tests AVRO processing.
+ * This test validates that KSML can produce AVRO messages and convert them to JSON using apicurio_avro notation.
+ */
+@Slf4j
+@Testcontainers
+class ApicurioAvroSchemaRegistryIT {
+
+    static Network network = Network.newNetwork();
+
+    @Container
+    static KafkaContainer kafka = new KafkaContainer("apache/kafka:4.0.0")
+            .withNetwork(network)
+            .withNetworkAliases("broker")
+            .withExposedPorts(9092, 9093);
+
+    @Container
+    static GenericContainer<?> schemaRegistry = new GenericContainer<>("apicurio/apicurio-registry:3.0.2")
+            .withNetwork(network)
+            .withNetworkAliases("schema-registry")
+            .withExposedPorts(8081)
+            .withEnv("QUARKUS_HTTP_PORT", "8081")
+            .withEnv("QUARKUS_HTTP_CORS_ORIGINS", "*")
+            .withEnv("QUARKUS_PROFILE", "prod")
+            .withEnv("APICURIO_STORAGE_KIND", "kafkasql")
```

Comment:
Why use the kafka version and not the in memory one?
Then it could start in parallel with Kafka instead of depending on Kafka.
Do we do persistent tests where the SR is restarted?